DesktopModules/Vanjaro/Core/Extensions/Workflow/Review/Managers/ModeratorManager.cs

```diff
@@ -92,14 +103,49 @@ public class ModeratorManager
             }
             Data.Add("Markup", Markup);
             Data.Add("WorkflowType", WorkflowManager.GetWorkflowType(wState.WorkflowID));
-            Data.Add("Page", Page);
+            Data.Add("ReviewContentInfo", rinfo);
             Data.Add("PreviousState", WorkflowManager.GetStateByID(WorkflowManager.GetPreviousStateID(wState.WorkflowID, wState.StateID)));
             Data.Add("CurrentState", wState);
             Data.Add("NextState", WorkflowManager.GetStateByID(WorkflowManager.GetNextStateID(wState.WorkflowID, wState.StateID)));
             Data.Add("LastState", WorkflowManager.GetStateByID(WorkflowManager.GetLastStateID(wState.WorkflowID).StateID));
             return Data;
         }
 
+     
+
+        public static void AddComment(string Entity, int EntityID, string Action, string Comment, PortalSettings PortalSettings)
+        {
+            List<Type> ExtensionTypes = new List<Type>();
+            string[] binAssemblies = Directory.GetFiles(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "bin")).Where(c => c.EndsWith(".dll")).ToArray();
+            foreach (string Path in binAssemblies)
+            {
+
+                try
+                {
+
+                    IEnumerable<Type> AssembliesToAdd = from t in Assembly.LoadFrom(Path).GetTypes()
+                                                        where typeof(IReviewComment).IsAssignableFrom(t) && t.GetConstructor(Type.EmptyTypes) != null
+                                                        select t;
+
+                    ExtensionTypes.AddRange(AssembliesToAdd.ToList());
+                }
+                catch { continue; }
+            }
+
+
+            foreach (Type t in ExtensionTypes)
+            {
+                IReviewComment ReviewContent = Activator.CreateInstance(t) as IReviewComment;
```

Comment:
This code does not belong in an extension. The code to load assemblies should be part of WorkflowManager and it needs to be cached.